### PR TITLE
Added Edit Employee command

### PR DESCRIPTION
**Pull request by kianhong95**

### CODE REVIEW

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        EmployeePosition updatedPosition = checkPosition(editEmployeeDescriptor.getPosition(), employeeToEdit.getPosition());
+
+        return new Employee(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedPosition);
+    }
+
+    /**
+    * Check for new phone value.
+    */
+    private static EmployeePhone checkPhone(EmployeePhone newEdit, EmployeePhone oldInfo){
+        if (newEdit.value == "00000000")
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: The "empty" values should be defined as static constants in the class responsible for it.
Ex: `public static final String EMPTY_EMPLOYEE_PHONE = "00000000";` in `EmployeePhone.java`

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            setEmail(toCopy.email);
+            setAddress(toCopy.address);
+            setPosition(toCopy.position);
+        }
+
+        public void setPhone(EmployeePhone phone) {
+            this.phone = phone;
+        }
+
+        public EmployeePhone getPhone() { return phone; }
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Expand this line into 3 lines

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+        public EmployeeEmail getEmail() {
+            return email;
+        }
+
+        public void setAddress(EmployeeAddress address) {
+            this.address = address;
+        }
+
+        public EmployeeAddress getAddress() { return address; }
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Expand this line into 3 lines

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+        public void setPosition(EmployeePosition position) {
+            this.position = position;
+        }
+
+        public EmployeePosition getPosition() { return position;}
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Expand this line into 3 lines

File: src/seedu/addressbook/commands/employee/EmployeeDeleteCommand.java [hunk 5]
```diff
@@ -2,19 +2,21 @@
...
+ * Deletes an employee identified using it's last displayed index from the Rms.
  */
 public class EmployeeDeleteCommand extends Command{
 
     public static final String COMMAND_WORD = "delemp";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n"
             + "Deletes the employee identified by the index number used in the last employee listing.\n\t"
-            + "Parameters: INDEX\n\t"
+            + "Parameters: INDEX (must be a positive integer)\n\t"
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeDeleteCommand.java:19]: The details and restrictions of inputs should only be declared in the user guide, unless it is not obvious

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        if (newEdit.value == "noargs")
+            return oldInfo;
+        return newEdit;
+    }
+
+    /**
+     * Stores the details to edit the employee with. Each non-empty field value will replace the
+     * corresponding field value of the employee.
+     */
+    public static class EditEmployeeDescriptor {
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Should be split into a different file

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+            return new CommandResult(Messages.MESSAGE_EMPLOYEE_NOT_IN_RMS);
+        }
+
+    }
+
+    /**
+     * Creates and returns an {@code Employee} with the details of {@code employeeToEdit}
+     * edited with {@code editEmployeeDescriptor}.
+     */
+    private static Employee createEditedEmployee(ReadOnlyEmployee employeeToEdit, EditEmployeeDescriptor editEmployeeDescriptor) {
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java:81]: These methods should be put in `Rms.java`, the `Command` class should only contain its relevant constants, its own constructor and the `execute()` method

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+        EmployeeAddress updatedAddress = checkAddress(editEmployeeDescriptor.getAddress(), employeeToEdit.getAddress());
+        EmployeePosition updatedPosition = checkPosition(editEmployeeDescriptor.getPosition(), employeeToEdit.getPosition());
+
+        return new Employee(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedPosition);
+    }
+
+    /**
+    * Check for new phone value.
+    */
+    private static EmployeePhone checkPhone(EmployeePhone newEdit, EmployeePhone oldInfo){
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java:95]: These static methods should be put in `Employee.java`

File: src/seedu/addressbook/parser/Parser.java [hunk 9]
```diff
@@ -301,7 +343,56 @@ private Command prepareEmployeeDelete(String args) {
...
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EmployeeEditCommand.MESSAGE_USAGE));
+        } catch (IllegalValueException ive) {
+        return new IncorrectCommand(ive.getMessage());
+        }
+    }
+
+    /**
+     * Returns new information to be edited if it is not empty
+     */
+    private static String checkEmpty(String toCheck, String argument){
```
Comment (px1099) [inline src/seedu/addressbook/parser/Parser.java]: Method name suggests that the method is a boolean method, so it should be changed

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+        if (newEdit.value == "00000000")
+            return oldInfo;
+        return newEdit;
+    }
+
+    /**
+     * Check for new email value.
+     */
+    private static EmployeeEmail checkEmail(EmployeeEmail newEdit, EmployeeEmail oldInfo){
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java:104]: These static methods should be put in Employee.java

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+        if (newEdit.value == "noargs@noargs.com")
+            return oldInfo;
+        return newEdit;
+    }
+
+    /**
+     * Check for new address value.
+     */
+    private static EmployeeAddress checkAddress(EmployeeAddress newEdit, EmployeeAddress oldInfo){
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java:113]: These static methods should be put in Employee.java

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+        if (newEdit.value == "noargs")
+            return oldInfo;
+        return newEdit;
+    }
+
+    /**
+     * Check for new position value.
+     */
+    private static EmployeePosition checkPosition(EmployeePosition newEdit, EmployeePosition oldInfo){
```
Comment (px1099) [inline src/seedu/addressbook/commands/employee/EmployeeEditCommand.java:122]: These static methods should be put in Employee.java

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java (hunk 1 above)
Comment (kianhong95) [reply src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Edited.

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java (hunk 2 above)
Comment (kianhong95) [reply src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Edited.

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java (hunk 3 above)
Comment (kianhong95) [reply src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Edited.

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java (hunk 4 above)
Comment (kianhong95) [reply src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Edited.

File: src/seedu/addressbook/commands/employee/EmployeeDeleteCommand.java (hunk 5 above)
Comment (kianhong95) [reply src/seedu/addressbook/commands/employee/EmployeeDeleteCommand.java:19]: Ok will add to user guide but will still leave the this in as the message will appear when user enters wrong command format.

File: src/seedu/addressbook/parser/Parser.java (hunk 9 above)
Comment (kianhong95) [reply src/seedu/addressbook/parser/Parser.java]: Edited.

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java (hunk 6 above)
Comment (kianhong95) [reply src/seedu/addressbook/commands/employee/EmployeeEditCommand.java]: Placed in a separate class file of its own.

File: src/seedu/addressbook/commands/employee/EmployeeEditCommand.java (hunk 7 above)
Comment (kianhong95) [reply src/seedu/addressbook/commands/employee/EmployeeEditCommand.java:81]: Is it fine if this createEditedEmployee is in its own EditEmployeeDescriptor class?

Comment (px1099) [reply src/seedu/addressbook/commands/employee/EmployeeEditCommand.java:81]: Yes, it is fine

File: src/seedu/addressbook/commands/employee/EmployeeDeleteCommand.java (hunk 5 above)
Comment (px1099) [reply src/seedu/addressbook/commands/employee/EmployeeDeleteCommand.java:19]: Ok, that is fine too

**Comment by px1099** [comment]
I think it is ready to be merged

**Pull request merged**